### ISSUE TITLE
Subframe selection causes MW4 in a loop

### ISSUE THREAD

**Issue by ACMBakx** [mworion/MountWizzard4#111]
[mw4-log%model.zip](https://github.com/mworion/MountWizzard4/files/5976282/mw4-log.model.zip)

Yesterday tested MW4 again.
Pointed at Betelgeuse with MW4 and with HC.
You have seen the result.
Did PA with the HC procedure and MC for a light model.
Next went back into MW4 to run a 75pts model.
To decrease the downloadtime I selected at subframe 90% to start with.
When I took a single image to check the downloadtime with this setting, MW4 got in a loop and continued to take images.
No response from cancel or bluntly starting the model run.
I had to quit MW4.
Restarted MW4 and started the modelrun with subframe at 100%.
Now ASTAP refused to solve any image.
Strange, because platesolve of a single frame did work, with ASTAP.
I did not change any settings.
Restarted the modelrun. Again ASTAP could not solve.
I then ran update.bat which installed a newer version.
Now MW4 modelbuilding did work correctly and build a nice model.
This problem with "subframe" has occurred before.
I cannot find it back in the attached log. I am sure you can.
I also attach the model that I custom build to accommodate my dome.
It should show the path I take. Top row first and then row after row zig-zag downwards.
Then the other side same. Like this the dome makes only small slews between points.

Best regards
Arie

**Comment by mworion** [comment]
Arie,
Thanks for the report and the log files. I take care of it. Right now I try to get some images taken as the weather is fine.
Michel

**Comment by mworion** [comment]
Hi Arie,
good news. After your first event, I added some logging. Now with the extended log I found the issue: I calculate the width / height and start position x,y for the subframe. This was done correct. When linked with binning, the spec says I have to calculate this in binned values -> also correct. Problem was the order of the commands: I set the binned mode at last and I had to do it at the beginning. So the camera driver saw a value, which was out of range and crashed. This also cause the loop.
At least that's my conclusion.
v1.0.3 will fix this point and will be released after test runs this evening.
Michel
